test/tests/functional/pbs_test_tpp.py

```diff
@@ -825,10 +920,162 @@ def test_COMM_ROUTERS_with_nondefault_pbs_conf(self):
                           job=True, interactive=True, resv=True,
                           resv_job=True)
 
+    def calculate_no_of_wait_threads(self, comm_pid):
+        """
+        This function calculate the number of wait threads
+        :param comm_pid: Comm's Pid
+        :type comm_pid: Integer.
+        """
+        ps_cmd = "ps -eT | grep pbs_comm | wc -l"
+        rc = self.server.du.run_cmd(self.server.hostname,
+                                    cmd=ps_cmd, as_script=True)
+        num_wait_threads = int(rc['out'][0]) - 1
+        return num_wait_threads
+
+    def test_PBS_COMM_THREADS(self):
+        """
+        Test allowable values for PBS_COMM_THREADS
+        """
+        threads = [1, 2, 100, 101, "T"]
+        for wait_thread in threads:
+            a = {'PBS_COMM_THREADS': wait_thread}
+            self.set_pbs_conf(host_name=self.mom.shortname, conf_param=a)
```

Comment:
'host_name' here is the hostname of first comm which in this suite we refer as self.server.hostname. Although it is the same as mom's hostname, it would be better to change it to self.server.hostname, which otherwise would depict that PBS_COMM_THREADS is getting set on a mom host.